src/provider/APIHandler.ts

```
@@ -282,8 +282,12 @@ export class APIHandler {
         }
     }
 
-    private reorderTabs(newOrdering: WindowIdentity[], tabId: ProviderIdentity): void {
-        const tab: DesktopWindow|null = this._model.getWindow(tabId as WindowIdentity);
+    private reorderTabs(newOrdering: WindowIdentity[], tabstrip: ProviderIdentity): void {
+        if (Array.isArray(newOrdering) && (!newOrdering || newOrdering.length === 0)) {
```

Will the isArray check ever be false? And if it can ever be false, I'm not sure about the logic here - 'if it's not an array, don't do any validation checks, but let it fail later rather than sooner'.

---

Worth renaming reOrderTabArray to reorderTabArray while you're here?

---

Fixed both comments - good catch on the logic!